Transaction Details – display headline currency in store's currency rather than shopper's currency

Jinksi opened the pull request:
Part of #7124

> **Warning**
> This PR is **NOT** behind a feature flag

#### Changes proposed in this Pull Request

This PR displays the Transaction Details headline payment amount in the store's currency when the store/charge currency differs (for multi-currency enabled stores).

[See design](https://www.figma.com/file/RiEQmaKRI7u9USAI3lyZz0VX/Dispute-Flows?node-id=5482%3A443300&mode=dev). 

> [!Note]
> This PR does not change the display currency of the new dispute summary amount – keeping feature-flag and non-feature-flag changes separate. See PR #7287.

#### Before/After

The screenshots below are for a transaction where:
* the store has multi-currency enabled
* the store is an Australian account with **AUD** set as the store's primary deposit currency
* the purchase was made with **USD** selected at checkout

**Before** – USD (charge currency) shows as headline amount

![image](https://github.com/Automattic/woocommerce-payments/assets/3147296/fe3b57e9-8488-4d38-bf93-1f90e50a92d2)

**After** – AUD (store currency) shows as headline amount
![image](https://github.com/Automattic/woocommerce-payments/assets/3147296/bdfee5fe-1459-449f-a364-8f648d21e786)

#### Testing instructions

<!--
Testing instructions: How should this be tested and how can a reviewer test the end-user functionality? Are there known issues that you plan to address in a future PR? Are there any side effects that readers should be aware of?
-->

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

1. Enable **Payments → Settings → Multi-currency**.
3. Visit **Payments → Overview** and note your store's deposit currency or currencies, visible as balance tabs.
4. Enable a non-store currency in **WooCommerce → Settings → Multi-currency**. Enable the currency switcher if using the Storefront theme (otherwise, show the front-end currency switcher via the block editor or another method). 
6. Create a transaction in (one of) the store's currency. The transaction details screen should show the headline payment amount in the store's currency. (example: a EUR charge currency and a EUR store currency)
    ![image](https://github.com/Automattic/woocommerce-payments/assets/3147296/03ccb626-a733-40ab-8ed1-8fc08f7adec2)
7. Create a transaction in a non-store currency using the currency switcher.
8. The transaction details screen should show the headline payment amount in the store's currency with the charge currency rendered below. (example: a JPY charge currency and a NOK store currency)
    ![image](https://github.com/Automattic/woocommerce-payments/assets/3147296/84157741-d193-4236-a199-ce9a67b0c3f4)

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

<!--
Make sure you edit the page for the current release when adding testing instructions.
We often create a blank page ahead of time for the next release.
If this PR need not be QA tested, edit to 'QA Testing Not Applicable'
-->

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

Jinksi committed:
Change 'Refunded' label to 'Deducted'

Jinksi committed:
Change 'Fee' label to 'Fees'

Jinksi committed:
Add missing `reporting_category` to BalanceTransaction type

in order to filter balance transactions by 'dispute' and get the dispute fee

Jinksi committed:
Add getDisputeFee util function

Jinksi committed:
Fix for ClickTooltip className prop not being applied

Jinksi committed:
Add tooltip content and styles

Jinksi committed:
Add test for tooltip fee breakdown

Jinksi committed:
Merge branch 'develop' into update/6974-transaction-details-breakdown-labels-and-fees

Jinksi committed:
Reorder imports to improve readability

Jinksi committed:
Update list item title styles to uppercase, match design

Jinksi committed:
Fix vertical alignment for list items with multiple content elements

e.g. text & ClickTooltip

Jinksi committed:
Add empty changelog entry

Jinksi committed:
Merge branch 'update/horizontal-list-label-style-uppercase' into update/6974-transaction-details-breakdown-labels-and-fees

Jinksi committed:
Update breakdown styles to uppercase, matching design

Jinksi committed:
Increase font sizes to improve text readability

Jinksi committed:
Add blank changelog entry

Jinksi committed:
Merge branch 'develop' into update/6974-transaction-details-breakdown-labels-and-fees

Jinksi committed:
Merge branch 'develop' into update/6974-transaction-details-breakdown-labels-and-fees

Jinksi committed:
Merge branch 'develop' into update/6974-transaction-details-breakdown-labels-and-fees

Jinksi committed:
Merge branch 'develop' into update/6974-transaction-details-breakdown-labels-and-fees

Jinksi committed:
Render store currency as headline

when charge currency differs from store currency

botwoo commented:
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `7159` or branch name `update/7124-transaction-details-headline-currency` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=update/7124-transaction-details-headline-currency) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: fad18787e5b55be191ce0ea530edc5ee61f9c925
- Build time: 2023-09-28 20:41:36 UTC

_Note: the build is updated when a new commit is pushed to this PR._

Jinksi committed:
Fix formatting of headline currency code

Jinksi committed:
Add aria-labels to payment amounts to improve a11y

Jinksi committed:
Add test case for different store/charge currency

Jinksi committed:
Add breakdown amount divider to match design

Jinksi committed:
Merge branch 'update/6974-transaction-details-breakdown-labels-and-fees' into update/7124-transaction-details-headline-currency

Jinksi committed:
Fix missing css closing bracket after merge conflict

shendy-a8c committed:
Merge branch 'develop' into update/6974-transaction-details-breakdown-labels-and-fees

Jinksi committed:
Merge branch 'develop' into update/6974-transaction-details-breakdown-labels-and-fees

Jinksi committed:
Updated non-feature-flag changelog entry

Jinksi committed:
Merge branch 'develop' into update/6974-transaction-details-breakdown-labels-and-fees

Jinksi committed:
Merge branch 'update/6974-transaction-details-breakdown-labels-and-fees' into update/7124-transaction-details-headline-currency

Jinksi committed:
Merge branch 'develop' into update/6974-transaction-details-breakdown-labels-and-fees

Jinksi committed:
Show "Deducted" fee label only for disputed transactions

Jinksi committed:
Add descriptive comment about auth/capture fee fallback

Jinksi committed:
Add explicit tests for Refunded/Deducted label logic

Jinksi committed:
Merge branch 'develop' into update/6974-transaction-details-breakdown-labels-and-fees

Jinksi committed:
Don't render fee breakdown tooltip for `won` dispute

Jinksi committed:
Merge branch 'update/6974-transaction-details-breakdown-labels-and-fees' into update/7124-transaction-details-headline-currency

shendy-a8c committed:
Merge branch 'develop' into update/6974-transaction-details-breakdown-labels-and-fees

Jinksi committed:
Merge branch 'develop' into update/6974-transaction-details-breakdown-labels-and-fees

Jinksi committed:
Ensure Transaction fee in tooltip is formatted in the fee currency

relevant for transactions where the charge currency differs from the fee balance transaction currency

Jinksi committed:
Rename transactionFee amount → fee for consistency

Jinksi committed:
Merge branch 'develop' into update/6974-transaction-details-breakdown-labels-and-fees

Jinksi committed:
Merge branch 'update/6974-transaction-details-breakdown-labels-and-fees' into update/7124-transaction-details-headline-currency

Jinksi committed:
Update snapshot test with added aria-label

haszari committed:
Merge branch 'develop' into update/6974-transaction-details-breakdown-labels-and-fees

Jinksi committed:
Merge branch 'develop' into update/6974-transaction-details-breakdown-labels-and-fees

Jinksi committed:
Merge branch 'develop' into update/6974-transaction-details-breakdown-labels-and-fees

Jinksi committed:
Merge branch 'update/6974-transaction-details-breakdown-labels-and-fees' into update/7124-transaction-details-headline-currency

Jinksi committed:
Merge branch 'develop' into update/6974-transaction-details-breakdown-labels-and-fees

Jinksi committed:
Remove redundant snapshot file after merge

Jinksi committed:
Fix duplicate import

Jinksi committed:
Add Dispute interface balance_transactions description

Jinksi committed:
Improve getDisputeFee logic to ignore if deduction reversed

Jinksi committed:
Prefer usage of getDisputeFeeFormatted

and rename getDisputeDeductedBalanceTransaction to clarify intent

Jinksi committed:
Update getDisputeFeeFormatted description

Jinksi committed:
Add further code comments to explain logic

Jinksi committed:
Merge branch 'update/6974-transaction-details-breakdown-labels-and-fees' into update/7124-transaction-details-headline-currency

Jinksi committed:
Merge branch 'develop' into update/7124-transaction-details-headline-currency

Jinksi committed:
Add changelog entry

Jinksi committed:
Merge branch 'develop' into update/7124-transaction-details-headline-currency

Jinksi committed:
Merge branch 'develop' into update/7124-transaction-details-headline-currency

haszari commented on client/payment-details/summary/index.tsx:
```diff
@@ -223,19 +223,40 @@ const PaymentDetailsSummary: React.FC< PaymentDetailsSummaryProps > = ( {
 					<div className="payment-details-summary__section">
-						<p className="payment-details-summary__amount">
+						<p
+							className="payment-details-summary__amount"
```
Should this class be changed / clarified since the value is different now (was shopper currency paid amount, is now merchant currency received amount).

haszari commented on client/payment-details/summary/index.tsx:
```diff
@@ -223,19 +223,40 @@ const PaymentDetailsSummary: React.FC< PaymentDetailsSummaryProps > = ( {
-									charge.amount,
-									charge.currency,
-									balance.currency
+								{ renderStorePrice ? (
```
Does this boolean mean "display headline number in merchant currency" or "show merchant currency label because shopper currency is different"?

Jinksi replied:
Answer: "show merchant currency label because shopper currency is different".

I've attempted to clarify this by naming the variable `isStoreAndShopperCurrencyMismatched` in fd93486.

haszari commented on client/payment-details/summary/index.tsx:
```diff
@@ -223,19 +223,40 @@ const PaymentDetailsSummary: React.FC< PaymentDetailsSummaryProps > = ( {
+									</>
 								) }
-								<span className="payment-details-summary__amount-currency">
-									{ charge.currency || 'USD' }
```
Nice to remove this potentially dangerous fallback to `USD`!

Jinksi replied:
I've also removed placeholder values (e.g. $0.00 USD) used for loading states in fd93486.

haszari commented on client/payment-details/summary/index.tsx:
```diff
@@ -246,10 +267,16 @@ const PaymentDetailsSummary: React.FC< PaymentDetailsSummaryProps > = ( {
 							{ renderStorePrice ? (
-								<p>
+								<p
+									className="payment-details-summary__breakdown__original-amount"
```
I'm thinking of this as "what the shopper paid" or shopper currency.

Jinksi replied:
Great idea 💡 – I've adjusted classNames and aria-labels to indicate "Payment amount in shopper's currency" instead of "Payment amount in original amount": cea9679.

haszari reviewed (approved):
Looks good, tests well, no blockers here 🚢 

I tested with feature flag off. My store has USD as the merchant/account currency, but my shopper was purchasing in NZD. I looked at three random transactions, and confirmed that the shopper value and merchant value swapped positions in the UI. 

This feels like a relatively important change (for multicurrency stores). What documentation do we need to update? Do we need to communicate this clearly to merchants & happiness? (Probably just add to What's New post)

Summarising my understanding of the change – let me know if I've misunderstood.

- The shopper currency value and merchant currency value are swapped in the transaction details UI
- The merchant currency value is now to the top / headline / large number, consistent with the transaction list screen
- This is only relevant to merchants that sell in different currencies (multicurrency stores)

Jinksi committed:
Merge branch 'develop' into update/7124-transaction-details-headline-currency

Jinksi committed:
Merge branch 'develop' into update/7124-transaction-details-headline-currency

Jinksi committed:
Merge branch 'develop' into update/7124-transaction-details-headline-currency

Jinksi committed:
Always use store currency for headline amount

(remove redundant ternary)

Jinksi committed:
Clarify the term "shopper" rather than "original" amount

Jinksi committed:
Indicate store's currency for headline amount aria-label

Jinksi committed:
Indicate store's currency for headline amount aria-label

Jinksi commented:
@haszari I've refactored some code and adjusted aria-labels in 1e93a2d, cea9679 and fd93486. Unit/integration tests are succeeding, but I'll conduct further manual tests to double-check that this PR behaves as expected.

> This feels like a relatively important change (for multicurrency stores). What documentation do we need to update? Do we need to communicate this clearly to merchants & happiness? (Probably just add to What's New post)

I'll check documentation and ensure updates are made if required. I think, for HEs, a descriptive What's new entry with screenshots will be appropriate.

I'll ensure critical flows are updated as well, if necessary.

> Summarising my understanding of the change – let me know if I've misunderstood.
* The shopper currency value and merchant currency value are swapped in the transaction details UI ✅
* The merchant currency value is now to the top / headline / large number, consistent with the transaction list screen ✅
* This is only relevant to merchants that sell in different currencies (multicurrency stores) ✅

haszari committed:
Merge branch 'develop' into update/7124-transaction-details-headline-currency

brucealdridge reviewed (commented):
I've had a look through the changes and they look good. 

My concern is that this PR could cause more confusion than it would solve. 
[When multi-currency is involved any fluctuations between the currencies will show up in the dispute](https://stripe.com/docs/disputes/how-disputes-work#disputed-amount). 

I sold an item for `40 EUR` on a USD store a month ago. Converted this was `$ 43.60 USD`. 
If the order of `40 EUR` was disputed today the amount is `$ 42.25`, this is great for the merchant but less so if the dispute is higher.

The timeline solves this well by listing the original and store currency and the conversion rate. 
![image](https://github.com/Automattic/woocommerce-payments/assets/57298/12c28816-02e1-4f75-881d-735f834b2553)

I wonder if by trying to hide some of the complexity for multicurrency merchants we are actually making important and relevant information less visible.

haszari commented:
> The timeline solves this well by listing the original and store currency and the conversion rate.

@brucealdridge Can we make changes to this PR to be more consistent with the timeline? 

I'm assuming the conversion only happens once, so we just need to ensure we always display the converted value using the correct historical conversion rate (again, assuming we have all this info!).

haszari commented:
Closing, we've decided not to implement this change (might be more confusing).

- https://github.com/Automattic/woocommerce-payments/issues/7124#issuecomment-1742273578
